core/src/main/java/io/questdb/jit/JitUtil.java

```diff
@@ -32,7 +32,19 @@ private JitUtil() {
     }
 
     public static boolean isJitSupported() {
-        // TODO what about FREEBSD_ARM64?
-        return Os.type != Os.LINUX_ARM64 && Os.type != Os.OSX_ARM64;
+        // TODO: what about FREEBSD_ARM64?
+        return Os.type != Os.LINUX_ARM64 &&
```

Comment:
This change disables JIT completely. If we merge it, JIT compiler will become a pile of unused code. What's the reason to disable it? The comment suggests that there is a leak in `NATIVE_JIT_LONG_LIST`. If you have a hypothesis that there is a leak, let's fix it (potentially in a separate PR) rather than disable the JIT compiler.

Reply:
Please let me know if you need any help with the investigation.

Reply:
Ok, so it is a leak in the compiler caused by this test:

```
sink.clear();
try (PreparedStatement ps = connection.prepareStatement("tab1 where ? is not null")) {
    ps.setString(1, "");
    try (ResultSet ignore1 = ps.executeQuery()) {
        Assert.fail();
    } catch (PSQLException e) {
        TestUtils.assertContains(e.getMessage(), "could not parse [value='', as=DOUBLE, index=0]");    }
}
```

This query is equiv. to "tab1 where ? != null" and the winning function (as per FunctionParser) is `EqDoubleFunctionFactory`, so setting the wildcard to a string is not accepted. But then rather than blow up nicely, some memory is leaked by CompiledFilterRecordCursorFactory.

This is hard to reproduce in my environment, so @bziobrowski said he could help.

I have reverted this, and now CI should fail.

I suppose I will look at why the type hint (string) is being overlooked.